src/cmake/externalpackages.cmake

```diff
@@ -348,7 +348,7 @@ endif ()
 
 option (BUILD_FMT_FORCE "Force local download/build of fmt even if installed" OFF)
 option (BUILD_MISSING_FMT "Local download/build of fmt if not installed" ON)
-set (BUILD_FMT_VERSION "6.1.2" CACHE STRING "Preferred fmtlib/fmt version, when downloading/building our own")
+set (BUILD_FMT_VERSION "7.0.1" CACHE STRING "Preferred fmtlib/fmt version, when downloading/building our own")
```

Comment:
I'm going to revert this line. I'm not 100% confident I've found every possible place I need to change to work around the "bool" bug, so I'll keep the default back at a safe version of fmt, and bump up just for some of the CI tests and for my own use until I'm sure it's fine.

Reply:
Do you think we should stick to 6.1.2 for the renderer as well?

Reply:
Did it pass the whole renderer and shading testsuites?

Reply:
Renderer yes, shading I don't know yet since we haven't done a build.

I'm not sure what kind of `sprintf` calls of `TypeDesc` going through fmtlib would be on a critical path, but it's hard to predict.

Reply:
So far I've just fixed the obvious bugs on the OIIO side, and I haven't yet fully explored what could break on the OSL side. That's next.

Reply:
Like, could it corrupt an oso file as it compiles?

Reply:
We compile plenty of shaders as part of the renderer testsuite, but its certainly worth seeing what the impact is on the OSL side.

Reply:
In any case - LGTM to this patch!